distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerProcessTests.java

```diff
@@ -361,6 +368,22 @@ public void testStop() throws Exception {
         assertThat(terminal.getErrorOutput(), containsString("final message"));
     }
 
+    public void testForceStop() throws Exception {
+        CountDownLatch blockMain = new CountDownLatch(1);
+        CountDownLatch inMain = new CountDownLatch(1);
+        mainCallback = (args, stdin, stderr, exitCode) -> {
+            stderr.println(SERVER_READY_MARKER);
+            inMain.countDown();
+            nonInterruptibleVoid(blockMain::await);
+        };
+        var server = startProcess(false, false);
+        nonInterruptibleVoid(inMain::await);
+        forceStopCallback = blockMain::countDown;
```

Comment:
Not sure I understand the intention of the `forceStopCallback` here, `nonInterruptibleVoid(blockMain::await)` will be interrupted and throw an AssertionError, counting down the latch doesn't make any difference.

Reply:
forceStop waits on the process to die. In this test, main is waiting indefinitely. Even though calling forceStop cancels the task, the thread is still alive, waiting on the latch. So without decrementing the latch, while in forceStop, the waitFor would wait indefinitely.